test/extended/operators/metrics.go

```diff
@@ -0,0 +1,144 @@
+package operators
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"k8s.io/client-go/kubernetes"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"k8s.io/client-go/dynamic"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	config "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+)
+
+var _ = g.Describe("OpenShift Operators", func() {
+
+	var (
+		clusterOperatorInfos []clusterOperatorInfo
+		kubeClient           kubernetes.Interface
+		dynamicClient        dynamic.Interface
+		serviceMonitorGvr    schema.GroupVersionResource = schema.GroupVersionResource{Group: "monitoring.coreos.com", Version: "v1", Resource: "servicemonitors"}
+		skip                 sets.String                 = sets.NewString(
+			"cluster-autoscaler",
+			"cloud-credential",
+			"cluster-autoscaler",
+			"image-registry",
+			"insights",
+			"machine-api",
+			"machine-config",
+			"marketplace",
+			"monitoring",
+			"network",
+			"node-tuning",
+			"openshift-samples",
+			"operator-lifecycle-manager",
+			"operator-lifecycle-manager-catalog",
+			"operator-lifecycle-manager-packageserver",
+			"service-ca",
+			"service-catalog-apiserver",
+			"service-catalog-controller-manager",
+			"storage",
+			"support",
+		)
```

Comment:
with a black list we are super far in terrain of the monitoring team. Either we switch to a whitelist of our operators, or we gift this to @s-urbaniak and @brancz.

Reply:
I would feel more comfortable with that approach also. There was a desire though to catch any "new" operator that did not get added to the whitelist, but I didn't find (yet) a reliable method of identifying "our" operators.

Reply:
@sanchezl The monitoring team does not maintain a list or a "registry" of operators that expose metrics in a given OpenShift installation. Setting up service monitors should be in the responsibility of a given operator itself and the correctness of that setup should be covered in e2e tests in the corresponding operator code base. I have doubts we should maintain such a list (be it blacklist or whitelist) in origin itself.

That also corresponds to the bug process. If exposing metrics for a given operator don't work because the service monitor setup is not correct, it is the concrete operator's responsibility to fix it.

/cc @smarterclayton @brancz